test/extensions/filters/http/ext_proc/ext_proc_integration_test.cc

```diff
@@ -0,0 +1,209 @@
+
+  void createUpstreams() override {
+    // Need to create a separate "upstream" for the gRPC server
+    HttpIntegrationTest::createUpstreams();
+    addFakeUpstream(FakeHttpConnection::Type::HTTP2);
+  }
+
+  void TearDown() override {
+    if (processor_connection_) {
+      AssertionResult result = processor_connection_->close();
+      RELEASE_ASSERT(result, result.message());
+      result = processor_connection_->waitForDisconnect();
+      RELEASE_ASSERT(result, result.message());
+    }
+    cleanupUpstreamAndDownstream();
+  }
+
+  void initializeConfig() {
+    config_helper_.addConfigModifier([this](envoy::config::bootstrap::v3::Bootstrap& bootstrap) {
+      // This is the cluster for our gRPC server, starting by copying an existing cluster
+      auto* server_cluster = bootstrap.mutable_static_resources()->add_clusters();
+      server_cluster->MergeFrom(bootstrap.static_resources().clusters()[0]);
+      server_cluster->set_name("ext_proc_server");
+      server_cluster->mutable_load_assignment()->set_cluster_name("ext_proc_server");
+      ConfigHelper::setHttp2(*server_cluster);
+
+      // Load configuration of the server from YAML and use a helper to add a grpc_service
+      // stanza pointing to the cluster that we just made
+      setGrpcService(*proto_config_.mutable_grpc_service(), "ext_proc_server",
+                     fake_upstreams_.back()->localAddress());
+
+      // Construct a configuration proto for our filter and then re-write it
+      // to JSON so that we can add it to the overall config
+      envoy::config::listener::v3::Filter ext_proc_filter;
+      ext_proc_filter.set_name("envoy.filters.http.ext_proc");
+      ext_proc_filter.mutable_typed_config()->PackFrom(proto_config_);
+      config_helper_.addFilter(MessageUtil::getJsonStringFromMessage(ext_proc_filter));
+      std::cerr << MessageUtil::getJsonStringFromMessage(bootstrap, true) << '\n';
+    });
+  }
+
+  envoy::extensions::filters::http::ext_proc::v3alpha::ExternalProcessor proto_config_{};
+  FakeHttpConnectionPtr processor_connection_;
+  FakeStreamPtr processor_stream_;
+};
+
+INSTANTIATE_TEST_SUITE_P(IpVersionsClientType, ExtProcIntegrationTest,
+                         GRPC_CLIENT_INTEGRATION_PARAMS);
+
+TEST_P(ExtProcIntegrationTest, GetAndCloseStream) {
+  initializeConfig();
+  setDownstreamProtocol(Http::CodecClient::Type::HTTP2);
+  HttpIntegrationTest::initialize();
+
+  auto conn = makeClientConnection(lookupPort("http"));
+  codec_client_ = makeHttpConnection(std::move(conn));
+  Http::TestRequestHeaderMapImpl headers;
+  HttpTestUtility::addDefaultHeaders(headers);
+  auto response = codec_client_->makeHeaderOnlyRequest(headers);
+
+  // Expect a message on the gRPC stream to the fake remote service
+  AssertionResult result =
+      fake_upstreams_.back()->waitForHttpConnection(*dispatcher_, processor_connection_);
+  RELEASE_ASSERT(result, result.message());
```

Comment:
For tests, prefer using `ASSERT_TRUE` (i.e. gtest asserts).

Reply:
Agreed. I followed a bum pattern from another test -- someone didn't read the docs on how to use AssertionResult in gtest.